### DISCUSSION TITLE
Add the ability to access the context into the resolveReference method for the Federation plugin

### DISCUSSION THREAD

**Discussion by moonflare** [hayes/pothos#341] [Q&A] [answered]
Hello there,

I am trying to setup the federation plugin, and when trying to define the first entity type, I ran into an issue where I'm not able to resolve the resource. In our app we use a pattern where we add different data loaders to the context, which we then access in the resolvers. Please see the example below ⬇️ 

```typescript
const MyType = builder.objectRef<MyType>('MyType').implement({
  fields: (t) => ({
    id: t.exposeID("id"),
    randomField: t.field({
      resolve: async (parent, _args, context) => {
        const randomField =
          await context.dataLoaders.randomFieldsSource.load(
            parent.randomFieldId
          );

        return randomField;
      },
      nullable: true,
      type: "MyTypeRandomField",
    }),
  }),
});
```

When setting up this type as an entity, I would like to be able to access that context and get the data from data loaders, something like this ⬇️ 

```typescript
builder.asEntity(MyType, {
  key: builder.selection<{ id: string }>('id'),
  resolveReference: (parent, _, context) => context.dataLoaders.myTypeSource.load(parent.id),
});
```

Currently, the argument I can access for the `resolveReference` method is `KeySelection` as defined [here](https://github.com/hayes/pothos/blob/53754982ad2cf3e3bbd7b6ce40926088a022ab3a/packages/plugin-federation/src/global-types.ts#L87). Would it be possible to extend this and add the possibility to access the args/context as well?

**Comment by hayes** [comment, marked as answer]
Yep, this should definitely be added.  I *think* this might not have been possible when I first wrote this plugin, but just checked, and federation definitely passes context now.  There won't be an args option, but the args should be (`parent`, `context`, `info`).  I'll update the types, but the args should already be passed to your resolve method, just not in the type definitions yet.

**Reply by moonflare** [reply]
Thanks for the quick reply @hayes 🙌🏻   Sounds good! Do you want me to create an issue for this?

**Reply by hayes** [reply]
Should be fixed already, just forgot to tell you
